shell/gpu/gpu_surface_metal_skia.mm

```diff
@@ -120,20 +122,9 @@
     return nullptr;
   }
 
-  // TODO(dnfield): Enable MSAA behind a flag if the settings say to.
-  // https://github.com/flutter/flutter/issues/100392
-  // Rough outline of steps needed:
-  // Create memoryless texture
-  // auto texture = drawable.get().texture;
-  // MTLTextureDescriptor* desc = [MTLTextureDescriptor texture2DDescriptorWithPixelFormat:
-  // texture.pixelFormat width: texture.width height: texture.height: mipmapped: texture.mipmapped];
-  // desc.sampleCount = 4;
-  // Type should be multisample, count 4 or 8 (4 always supported)
-  // Give the new texture to skia in this call, using the same sampleCount value
-  // Resolve texture
   auto surface = CreateSurfaceFromMetalTexture(context_.get(), drawable.get().texture,
                                                kTopLeft_GrSurfaceOrigin,  // origin
-                                               1,                         // sample count
+                                               msaa_samples_,             // sample count
```

Comment:
Some validation of sample counts would be good here. [Docs say it could be 1, 2, 4 or 8](https://developer.apple.com/documentation/metal/mtldevice/1433355-supportstexturesamplecount?language=objc). Though we should stick to 1 and 4 for max compatibility.

Reply:
The `Settings` object validates that this is `0, 1, 2, 4, 8, or 16`. I can add tighter validation here, but should it just bea DCHECK?